Sources/MapboxDirections/DirectionsError.swift

```diff
@@ -8,7 +8,11 @@ import FoundationNetworking
  */
 public enum DirectionsError: LocalizedError {
     
-    public init(code: String?, message: String?, response: URLResponse?, underlyingError error: Error?) {
+    public init(code: String?, message: String?, response: URLResponse?, underlyingError error: Error?, refreshTTL: Int? = nil) {
+        guard refreshTTL == nil || refreshTTL != 0 else {
```

Comment:
Are we sure that in the case when `refreshTTL  == 0` and e.g. `response.statusCode == 404` the error should be `. refreshExpired`? 
As for me, it's better first to check the non-200 response code and only then check `refreshTTL`. 
Btw, how does the Android platform handle this case?

Reply:
`refresh_ttl` in the refresh response will be appended and =0 only in case of the refresh invalidation error. Android also does [the check](https://github.com/mapbox/mapbox-navigation-android/pull/7318/files#diff-0fd6e55eb977c007ca6298a2d0610352b533bd1166f2c0e6eb5571b7cb4cb8ecR164) for this parameter to be 0